src/switch_impl_intf.ml

```diff
@@ -2,11 +2,13 @@ open Import
 
 let insert_inferred_intf ~source_uri client text_editor =
   let open Promise.Syntax in
-  (* XXX this seems sketchy. shouldn't it work for reason as well? *)
-  match String.is_suffix source_uri ~suffix:".ml" with
+  match
```

Comment:
what do you think about making a string set of suffixes and checking for presence based on `Filename.extension`?

That could easily support adding e.g. `mlx` or upcoming syntaxes just by adding a string to a set.

Reply:
Clean-up should be done separately from this PR. But good point!